sentinel-transport/sentinel-transport-spring-mvc/src/main/java/com/alibaba/csp/sentinel/transport/command/SentinelApiHandler.java

```diff
@@ -0,0 +1,110 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.transport.command;
+
+import com.alibaba.csp.sentinel.command.CommandHandler;
+import com.alibaba.csp.sentinel.command.CommandRequest;
+import com.alibaba.csp.sentinel.command.CommandResponse;
+import com.alibaba.csp.sentinel.config.SentinelConfig;
+import com.alibaba.csp.sentinel.transport.command.http.StatusCode;
+import com.alibaba.csp.sentinel.transport.log.CommandCenterLog;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.PrintWriter;
+import java.util.Map;
+
+/**
+ * @author shenbaoyong
+ */
+public class SentinelApiHandler {
+
+    public static final String SERVER_ERROR_MESSAGE = "Command server error";
+    public static final String INVALID_COMMAND_MESSAGE = "Invalid command";
+
+    private CommandHandler commandHandler;
+
+    public SentinelApiHandler(CommandHandler commandHandler) {
+        this.commandHandler = commandHandler;
+    }
+
+    public void handle(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
+        PrintWriter printWriter = null;
+        try {
+            long start = System.currentTimeMillis();
+            printWriter = httpServletResponse.getWriter();
+            CommandCenterLog.info("[SentinelApiHandler] request income: " + httpServletRequest.getRequestURL());
+            CommandRequest request = new CommandRequest();
+            Map<String, String[]> parameterMap = httpServletRequest.getParameterMap();
+            for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
+                String[] value = entry.getValue();
+                if (value != null && value.length >= 1) {
+                    request.addParam(entry.getKey(), value[0]);
+                }
+            }
+            if (commandHandler == null) {
+                writeResponse(httpServletResponse, printWriter, StatusCode.BAD_REQUEST, INVALID_COMMAND_MESSAGE);
+                return;
+            }
+            CommandResponse<?> response = commandHandler.handle(request);
+            handleResponse(response, httpServletResponse, printWriter);
+
+            long cost = System.currentTimeMillis() - start;
+            CommandCenterLog.info("[SentinelApiHandler] Deal request : " + httpServletRequest.getRequestURL()
+                     + ", time cost: " + cost + " ms");
+        } catch (Throwable e) {
+            CommandCenterLog.warn("[SentinelApiHandler] error", e);
+            try {
+                if (printWriter != null) {
+                    String errorMessage = SERVER_ERROR_MESSAGE;
```

Comment:
An unnecessary local variable here.

Reply:
yes I will change it to  
```java
writeResponse(httpServletResponse, printWriter, StatusCode.INTERNAL_SERVER_ERROR, SERVER_ERROR_MESSAGE);
```

Reply:
> yes I will change it to
> 
> ```java
> writeResponse(httpServletResponse, printWriter, StatusCode.INTERNAL_SERVER_ERROR, SERVER_ERROR_MESSAGE);
> ```

Loog forward to your improvement, Baoyong